### PR TITLE
Homework 1

### PR DESCRIPTION
**Pull request by prashant1shukla**

### CODE REVIEW

**Comment by prashant1shukla** [comment]
| Metric | Value |
| ------ | ----- |
| alert_status | <b>:white_check_mark: OK</b> |
| bugs | 0 |
| code_smells | 26 |
| reliability_rating | A (0 Bugs)<br/>Note: A being best, E being worst |
| security_rating | A (0 Vulnerabilities)<br/>Note: A being best, E being worst |
| vulnerabilities | 0 |

File: backend/homework/1/app/src/main/java/org/example/Main.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+            grade=sc.next().charAt(0);
+            st.setGrade(grade);
+
+            students.addStudent(st);
+        }
+
+        String id="id01";
+        Student st2=students.getStudent(id);
+        System.out.println("The attributes of required student are: 1.Name: "+ st2.getName()+" 2.Age: "+st2.getAge()+" 3.Grade: "+st2.getGrade());
+
```
Comment (anupam-srivastava-kd) [inline backend/homework/1/app/src/main/java/org/example/Main.java:43]: doesn't using toString method seems practical here?

File: .github/workflows/maven-sonar.yml [hunk 2]
```diff
@@ -0,0 +1,58 @@
+name: Maven Sonar Analysis
```
Comment (anupam-srivastava-kd) [inline .github/workflows/maven-sonar.yml:1]: Update pr description, also why did you merge the pr??

File: backend/homework/1/app/src/main/java/org/example/Main.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+            String name;
+            name=sc.nextLine();
+            st.setName(name);
+
+            System.out.println("Enter the Student Age: ");
+            int age;
+            age=sc.nextInt();
+            st.setAge(age);
+
+            System.out.println("Enter the Student Grade: ");
```
Comment (anupam-srivastava-kd) [inline backend/homework/1/app/src/main/java/org/example/Main.java:32]: Why are we using sout here, should have used comments to explain this

File: backend/homework/1/core/src/main/java/org/example/Student.java [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+public class Student {
+    private String id;
+    private String name;
+    private int age;
+    private  char grade;
+    public String getId(){
+        return id;
+    }
+    public void setId(String id){
```
Comment (anupam-srivastava-kd) [inline backend/homework/1/core/src/main/java/org/example/Student.java:11]: shouldn't we use all args constructor

File: backend/homework/1/data/src/main/java/org/example/StudentRepository.java [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+    public static ArrayList<Student> students=new ArrayList<>();
+    private static final Logger slf4jLogger = LoggerFactory.getLogger(LogBack.class);
+
+    public void addStudent(Student x){
+        students.add(x);
+        slf4jLogger.info("Your Student is Added");
+    }
+
+    public Student getStudent(String id){
+        for(int i=0;i<students.size();i++){
```
Comment (anupam-srivastava-kd) [inline backend/homework/1/data/src/main/java/org/example/StudentRepository.java:17]: use meaningful names

Comment (anupam-srivastava-kd) [reply backend/homework/1/data/src/main/java/org/example/StudentRepository.java:17]: Check corner cases and add checks

File: backend/homework/1/logging/src/main/java/org/example/LogBack.java [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+import org.slf4j.LoggerFactory;
+
+public class LogBack {
+    private static final Logger slf4jLogger = LoggerFactory.getLogger(LogBack.class);
+
+    public static void main(String[] args) {
+
+        slf4jLogger.info("Welcome to the first homework of Backend done by Prashant");
+
+    }
```
Comment (anupam-srivastava-kd) [inline backend/homework/1/logging/src/main/java/org/example/LogBack.java:13]: use log levels, there is no use of using a single line for an entire module.

**Pull request merged**